ingraph-engine/ingraph-engine-ingraph-ire/src/main/scala/ingraph/ire/Indexer.scala

```diff
@@ -16,6 +18,11 @@ case class IngraphVertex(id: Long,
   val reverseEdges: mutable.ListMap[String, IngraphEdge] = mutable.ListMap[String, IngraphEdge]()
   override def toString: String = s"Vertex($id, $properties)"
 }
+object IngraphVertex {
+  implicit val ingraphObject = new IngraphObject[IngraphVertex] {
```

Comment:
I followed the `typeclass` idiom instead of inheritance similar to that in Haskell or Rust.
This is the 

- implementation (called `impl` in Rust, `instance` in Haskell)
- of the trait `IngraphObject` (called `trait` in Rust and Scala, `class` in Haskell)
- for the structure `IngraphVertex` called `data` (Haskell), `struct` in (Rust).

Example for using a typeclass instead of supertype:
```scala
  import ingraph.ire.IngraphObject
  object elementPropertyExists {
    def apply[T : IngraphObject](obj: T, key: String): Option[AnyRef] = {
      implicitly[IngraphObject[T]].properties(obj) get key
    }
    // example usage:
    import ingraph.ire.IngraphVertex
    val xyz = apply(IngraphVertex(0, Set(), Map("abc" -> "def")), "abc")
  }

```